From b20c08befd9867c3cf0ffa84a5d3067f95911ee1 Mon Sep 17 00:00:00 2001
From: Yuanjie Huang <Yuanjie.Huang@windriver.com>
Date: Thu, 23 Jul 2015 09:01:07 +0000
Subject: [PATCH] powerpc/85xx: Use KCFLAGS when looking for libgcc

Freescale SDK kernel v1.6 and above adds code with dependency on libgcc
to support Freescale DPAA Q/Bman, but the makefile change has not
considered that a cross-compiler normally needs an sysroot option to
correctly report the path of libgcc.

Since CC variable is composed from CROSS_COMPILE variable in top-most
makefile, the sysroot has to be set using KCFLAGS, which is designed to
provide additional options to the C compiler. Moreover, KCFLAGS is not
appended to KBUILD_CFLAGS propagated to architecture specific makefile.
Therefore, this patch modifies the compiler command for querying libgcc
path to use KCFLAGS option, so that once --sysroot=<path> option is set
in KCFLAGS, libgcc can be found correctly.

Signed-off-by: Yuanjie Huang <Yuanjie.Huang@windriver.com>
---
 arch/powerpc/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index 070b78a..b66f453 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -96,7 +96,7 @@ LDFLAGS_vmlinux-y := -Bstatic
 LDFLAGS_vmlinux-$(CONFIG_RELOCATABLE) := -pie
 LDFLAGS_vmlinux	:= $(LDFLAGS_vmlinux-y)
 
-LIBGCC          := $(shell $(CC) $(KBUILD_CFLAGS) -print-libgcc-file-name)
+LIBGCC          := $(shell $(CC) $(KBUILD_CFLAGS) $(KCFLAGS) -print-libgcc-file-name)
 
 ifeq ($(CONFIG_PPC64),y)
 ifeq ($(call cc-option-yn,-mcmodel=medium),y)
-- 
1.7.5.4

